### PR TITLE
Add manifest validation, replace manifest build dir generation

### PR DESCRIPTION
**Pull request by KarenkovID**
Added 2 flags to forma configuration for working with manifests:
1. Remove `generateManifest` from android targets and add `generateMissedManifests` to forma global config
2. Add `validateManifestPackages` for comparing build.gradle.kts `packageName` with manifest's `package`

### CODE REVIEW

File: plugins/android/src/main/java/tools/forma/android/feature/AndroidLibrary.kt [hunk 1]
```diff
@@ -94,9 +113,7 @@ private fun populateManifest(
             parentFile.mkdirs()
             createNewFile()
             writeText(
-                """<?xml version="1.0" encoding="utf-8"?>
-                           <manifest package="$packageName"/>
-                """.trimIndent()
+                """<manifest package="$packageName"/>"""
```
Comment (stepango) [inline plugins/android/src/main/java/tools/forma/android/feature/AndroidLibrary.kt:116]: Why we are skipping xml version here?

File: plugins/android/src/main/java/tools/forma/android/feature/AndroidLibrary.kt [hunk 2]
```diff
@@ -57,24 +60,40 @@ fun androidLibraryFeatureDefinition(
...
+        }
+    }
+}
+
 private fun LibraryExtension.maybeGenerateManifest(
     project: Project,
     feature: AndroidLibraryFeatureConfiguration
 ) {
-    val manifestFile = manifestFile(project.buildDir, feature.packageName)
+    val manifestFile = File(project.projectDir, "src/main/AndroidManifest.xml")
```
Comment (stepango) [inline plugins/android/src/main/java/tools/forma/android/feature/AndroidLibrary.kt:93]: There could be a few cases where we'll need different manifests in `src/smth` folders do we want to support that?

**Review by stepango — APPROVED** [review]
Looks great, my only worry is xml version and encoding info which we are removing from the manifest. I wonder what's the implications would be in this case.

File: plugins/android/src/main/java/tools/forma/android/feature/AndroidLibrary.kt (hunk 1 above)
Comment (KarenkovID) [reply plugins/android/src/main/java/tools/forma/android/feature/AndroidLibrary.kt:116]: I've never seen any effect of specifying xml version in android xml files like manifest or resources, I can roll back this

**Review by michaem — APPROVED** [review]

**Pull request merged**